components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/RemoteFetchConfigurationServiceImpl.java

```diff
@@ -166,4 +170,55 @@ public void triggerRemoteFetch(RemoteFetchConfiguration fetchConfiguration) thro
                     fetchConfiguration.getRemoteFetchConfigurationId());
         }
     }
+
+    /**
+     * Validate limit.
+     * Check optionalLimit has a value, or else set to default value.
+     * @param optionalLimit given limit value.
+     * @return validated limit and offset value.
+     */
+    private int validateLimit(OptionalInt optionalLimit) throws RemoteFetchCoreException {
+
+        int limit = optionalLimit.orElse(RemoteFetchConfigurationUtils.getDefaultItemsPerPage());
```

Comment:
This has to go after the 
if (log.isDebugEnabled()) {

as we are supposed to do it after logging.